src/test/java/net/snowflake/client/jdbc/CallableStatementIT.java

```diff
@@ -60,54 +60,55 @@ public static Connection getConnection() throws SQLException {
   private final String deleteStoredProcedure = "drop procedure if exists square_it(FLOAT)";
   private final String deleteSecondStoredProcedure = "drop procedure if exists add_nums(INT, INT)";
 
-  protected Connection connection = null;
-  protected Statement statement = null;
-
   @Before
   public void setUp() throws SQLException {
-    Connection con = getConnection();
-    statement = con.createStatement();
-    statement.execute(createStoredProcedure);
-    statement.execute(createSecondStoredProcedure);
-    con.close();
+    try (Connection con = getConnection();
+        Statement statement = con.createStatement()) {
+      statement.execute(createStoredProcedure);
+      statement.execute(createSecondStoredProcedure);
+    }
   }
 
   @After
   public void tearDown() throws SQLException {
-    Connection con = getConnection();
-    statement = con.createStatement();
-    statement.execute(deleteStoredProcedure);
-    statement.execute(deleteSecondStoredProcedure);
-    con.close();
+    try (Connection con = getConnection();
+        Statement statement = con.createStatement()) {
+      statement.execute(deleteStoredProcedure);
+      statement.execute(deleteSecondStoredProcedure);
+    }
   }
 
   @Test
   public void testPrepareCall() throws SQLException {
     // test CallableStatement with no binding parameters
-    connection = getConnection();
-    statement = connection.createStatement();
-    CallableStatement callableStatement = connection.prepareCall("call square_it(5)");
-    assertThat(callableStatement.getParameterMetaData().getParameterCount(), is(0));
-
-    // test CallableStatement with 1 binding parameter
-    callableStatement = connection.prepareCall("call square_it(?)");
-    // test that getParameterMetaData works with CallableStatement. At this point, it always returns
-    // the type as "text."
-    assertThat(callableStatement.getParameterMetaData().getParameterType(1), is(Types.VARCHAR));
-    callableStatement.getParameterMetaData().getParameterTypeName(1);
-    assertThat(callableStatement.getParameterMetaData().getParameterTypeName(1), is("text"));
-    callableStatement.setFloat(1, 7.0f);
-    ResultSet rs = callableStatement.executeQuery();
-    rs.next();
-    assertEquals(49.0f, rs.getFloat(1), 1.0f);
-
-    // test CallableStatement with 2 binding parameters
-    callableStatement = connection.prepareCall("call add_nums(?,?)");
-    callableStatement.setDouble(1, 32);
-    callableStatement.setDouble(2, 15);
-    rs = callableStatement.executeQuery();
-    rs.next();
-    assertEquals(47, rs.getDouble(1), .5);
+    try (Connection connection = getConnection()) {
+      try (CallableStatement callableStatement = connection.prepareCall("call square_it(5)")) {
```

Comment:
Make sense. What I am worried about is that other tests also reuse one variable over 10 times when they create different resultsets. Do you want to change them as well?

Reply:
I think we do not need to merge two different try blocks in this case because the callabeStatement can be closed before the different callableStatment is initiated.

Reply:
Sure that makes sense. We can leave this as-is.